src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides length must by > 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a more detailed description in the error message. A better message would be: 'The provided sides do not form a valid triangle as one side is equal to or greater than the sum of the other two.'